tests/test_packages/test_skills/test_generic_buyer/test_behaviours.py

```diff
@@ -0,0 +1,381 @@
+        )
+
+        # before
+        assert self.get_quantity_in_outbox() == 0
+
+        # operation
+        search_behaviour.act()
+
+        # after
+        assert self.get_quantity_in_outbox() == 0
+
+
+class TestSkillHandler(BaseSkillTestCase):
+    """Test handlers of generic buyer."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "generic_buyer")
+
+    def test_fipa_handler_handle_unidentified_dialogue(self):
+        """Test the _handle_unidentified_dialogue method of the fipa handler."""
+        # setup
+        fipa_handler = cast(GenericFipaHandler, self.skill.skill_context.handlers.fipa)
+        incoming_message = self.build_incoming_message(
+            message_type=FipaMessage,
+            dialogue_reference=("1", "0"),
+            performative=FipaMessage.Performative.ACCEPT,
+        )
+
+        # before
+        assert self.get_quantity_in_outbox() == 0
+
+        # operation
+        fipa_handler._handle_unidentified_dialogue(cast(FipaMessage, incoming_message))
+
+        # after
+        assert self.get_quantity_in_outbox() == 1
+        res_bool, res_str = self.message_has_attributes(
+            actual_message=self.get_message_from_outbox(),
+            message_type=DefaultMessage,
+            performative=DefaultMessage.Performative.ERROR,
+            to=incoming_message.sender,
+            sender=self.skill.skill_context.agent_address,
+            error_code=DefaultMessage.ErrorCode.INVALID_DIALOGUE,
+            error_msg="Invalid dialogue.",
+            error_data={"fipa_message": incoming_message.encode()},
+        )
+        assert res_bool is True, res_str
+
+    def test_fipa_handler_handle_propose(self):
+        """Test the _handle_propose method of the fipa handler."""
+        # ToDo need to mock affordable and acceptable values
+        # setup
+        fipa_handler = cast(GenericFipaHandler, self.skill.skill_context.handlers.fipa)
+        proposal = Description(
+            {
+                "ledger_id": "some_ledger_id",
+                "price": "some_price",
+                "currency_id": "some_currency_id",
+                "service_id": "some_service_id",
+                "quantity": "some_quantity",
+                "tx_nonce": "some_tx_nonce",
+            }
+        )
+        incoming_message = self.build_incoming_message(
+            message_type=FipaMessage,
+            dialogue_reference=("1", "0"),
+            message_id=2,
+            target=1,
+            performative=FipaMessage.Performative.PROPOSE,
+            proposal=proposal,
+        )
+
+        # before
+        assert self.get_quantity_in_outbox() == 0
+
+        # operation
+        fipa_handler._handle_propose(
+            cast(FipaMessage, incoming_message),
+            FipaDialogue(
+                DialogueLabel(("1", "1"), "counterparty", "me"),
+                "me",
+                FipaDialogue.Role.BUYER,
+            ),
+        )
```

Comment:
If you feed the right arguments (i.e. the correctly defined incoming message) to the `handle` method then the skill will take care of this. Don't repeat work the skill already does. So instead, create a propose like message and feed it to the `handle` method. To ensure the skill's dialogue is at the correct stage, we can either mock the skill's dialogue or manually "prep" it with another helper method.

Reply:
Yep, I am working on the quick dialogue builder.